extensions/database-documentation/src/common/utils.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata'
+import * as vscode from 'vscode';
+// Language localization features
+import * as nls from 'vscode-nls';
+// The module 'openai' contains the OpenAI API
+import { Configuration, OpenAIApi } from "openai";
+
+const localize = nls.loadMessageBundle();
+
+export async function generateMarkdown(context: azdata.ObjectExplorerContext): Promise<string> {
+	let connection = (await azdata.connection.getCurrentConnection());
+	let databaseName = context.connectionProfile.databaseName;
+
+	let tables: string[];
+	if (context.nodeInfo.nodeType === 'Table') {
+		tables = [context.nodeInfo.metadata.name];
+	}
+	else {
+		const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>("MSSQL", azdata.DataProviderType.QueryProvider);
+		const connectionUri = await azdata.connection.getUriForConnection(connection.connectionId);
+		let query = `SELECT TABLE_NAME FROM [${databaseName}].INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = 'BASE TABLE'`;
```

Comment:
Make sure you're escaping `]` and `'` whenever you manual create T-SQL statements like this to prevent SQL injection type issues. (easy way is to test with a DB that has both those characters in its name)

https://learn.microsoft.com/en-us/sql/relational-databases/security/sql-injection?view=sql-server-ver16

Reply:
make sure this comment is addressed.